src/client.rs

```diff
@@ -1535,117 +1406,35 @@ mod test {
 
     async fn test_versioned_init(consistent_snapshot: bool) {
         let repo = EphemeralRepository::<Json>::new();
-        let mut remote = Repository::new(&repo);
-
-        //// First, create the root metadata.
-        let root1 = RootMetadataBuilder::new()
-            .consistent_snapshot(consistent_snapshot)
-            .version(1)
-            .expires(Utc.ymd(2038, 1, 1).and_hms(0, 0, 0))
-            .root_key(KEYS[0].public().clone())
-            .snapshot_key(KEYS[0].public().clone())
-            .targets_key(KEYS[0].public().clone())
-            .timestamp_key(KEYS[0].public().clone())
-            .signed::<Json>(&KEYS[0])
-            .unwrap();
 
-        let mut root2 = RootMetadataBuilder::new()
-            .consistent_snapshot(consistent_snapshot)
-            .version(2)
-            .expires(Utc.ymd(2038, 1, 1).and_hms(0, 0, 0))
-            .root_key(KEYS[1].public().clone())
-            .snapshot_key(KEYS[1].public().clone())
-            .targets_key(KEYS[1].public().clone())
-            .timestamp_key(KEYS[1].public().clone())
-            .signed::<Json>(&KEYS[1])
-            .unwrap();
-
-        // Make sure the version 2 is signed by version 1's keys.
-        root2.add_signature(&KEYS[0]).unwrap();
-
-        let mut targets = TargetsMetadataBuilder::new()
-            .signed::<Json>(&KEYS[0])
-            .unwrap();
-
-        targets.add_signature(&KEYS[1]).unwrap();
-
-        let mut snapshot = SnapshotMetadataBuilder::new()
-            .insert_metadata(&targets, &[HashAlgorithm::Sha256])
-            .unwrap()
-            .signed::<Json>(&KEYS[0])
-            .unwrap();
-
-        snapshot.add_signature(&KEYS[1]).unwrap();
-
-        let mut timestamp =
-            TimestampMetadataBuilder::from_snapshot(&snapshot, &[HashAlgorithm::Sha256])
-                .unwrap()
-                .signed::<Json>(&KEYS[0])
-                .unwrap();
-
-        timestamp.add_signature(&KEYS[1]).unwrap();
-
-        ////
-        // Now register the metadata (root version 1 and 2).
-        let root_path = MetadataPath::from_role(&Role::Root);
-        let targets_path = MetadataPath::from_role(&Role::Targets);
-        let snapshot_path = MetadataPath::from_role(&Role::Snapshot);
-        let timestamp_path = MetadataPath::from_role(&Role::Timestamp);
-
-        remote
-            .store_metadata(
-                &root_path,
-                &MetadataVersion::Number(1),
-                &root1.to_raw().unwrap(),
-            )
-            .await
-            .unwrap();
-
-        remote
-            .store_metadata(&root_path, &MetadataVersion::None, &root1.to_raw().unwrap())
-            .await
-            .unwrap();
-
-        let metadata_version = if consistent_snapshot {
```

Comment:
is a repository in consistent_snapshots mode supposed to write unversioned targets/snapshots?  A functional change with RepoBuilder is that it always writes the unversioned files, but I'm not sure if that's a bad thing.  I suppose if the client were to start incorrectly reading the unversioned files, this test wouldn't catch it anymore.